### PR TITLE
OOP school library: refactor your code

### PR DESCRIPTION
**Pull request by udeaghad**
In this pull request, we have paid special attention to the "Single-responsibility Principle" by

- ensuring that the role of main.rb is to start our code and ask the user for the initial choice of options (or to quit).
- Extracting some pieces of logic into a separate class(es).

Kindly review and approve.

### CODE REVIEW

File: create.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  end
+
+  def create_student
+    puts 'Add Student'
+    puts 'Enter Student Age: '
+    age = gets.chomp.to_i
+    puts 'Enter Student Name: '
+    name = gets.chomp
+    puts 'Parent Permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
```
Comment (Bondok6) [inline create.rb]: - [ ] The Single-Responsibility Principle of the `SOLID` object Oriented Design states that a class should have one job only. hence all requests for inputs should have been placed in the `main.rb` file /or `classes/methods` whose responsibilities are to request input. Please kindly work on this and move a step forward toward approval. 👌

  For more on the SOLID lessons, click on this [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) 💡

File: create.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  end
+
+  def create_teacher
+    puts 'Add Teacher'
+    print 'Enter Teacher Age: '
+    age = gets.chomp.to_i
+    print 'Enter Teacher Name: '
+    name = gets.chomp
+    print 'Enter Teacher Specialization: '
+    specialization = gets.chomp
```
Comment (Bondok6) [inline create.rb]: - [ ] As I mentioned above ⬆ All requests for inputs should have been placed in the main.rb file or classes/methods whose responsibilities are to request input for reasons outlined above.👌

File: create.rb [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    @persons.push(teacher)
+    puts '1 Teacher Added'
+  end
+
+  def create_book
+    puts 'Add Book'
+    puts 'Enter Title: '
+    title = gets.chomp
+    puts 'Enter Author: '
+    author = gets.chomp
```
Comment (Bondok6) [inline create.rb]: - [ ] As I mentioned above ⬆ All requests for inputs should have been placed in the main.rb file or classes/methods whose responsibilities are to request input for reasons outlined above. 👌

> For example you can make a file called `require_inputs.rb` containing some functions responsible for requesting input from users 💡
```
// require_inputs.rb

def grap_book_data
    puts 'Add Book'
    puts 'Enter Title: '
    title = gets.chomp
    puts 'Enter Author: '
    author = gets.chomp

  [title, author]
end
```
```
// create.rb

def create_book
  title, author = grap_book_data
  book = Book.new(title, author)
  @books.push(book)
  puts "Book by #{author} is created."
end
```

**Review by Bondok6 — CHANGES_REQUESTED** [review]
## Hi @udeaghad and @efoosag 🙋‍♂️, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights
- No Linter errors ✅
- Your project is professional ✅
- Using GitHub flow ✅

## Required Changes ♻️
- [ ] I noticed that the link of your MIT.md file in your README is broken, so I suggest adding the `MIT.md` file to your project 👌, Here is the [Micorverse Template](https://github.com/microverseinc/readme-template/blob/master/MIT.md), and remember to customize it 😉

> The MIT license gives users express permission to reuse code for any purpose, sometimes even if the code is part of proprietary software. As long as users include the original copy of the MIT license in their distribution, they can make any changes or modifications to the code to suit their own needs. 💯

- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by dasileker — APPROVED** [review]
### **Hi @efoosag,** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**